### DISCUSSION TITLE
Welcome to ChatGPT Code Interpreter Experiments 🧪 Discussions!

### DISCUSSION THREAD

**Discussion by SkalskiP** [SkalskiP/awesome-chatgpt-code-interpreter-experiments#4] [Announcements]
## 👋 Welcome!

ChatGPT Code Interpreter is, in my opinion, the most interesting, thing happening in AI today. Here we can exchange ideas, discuss exploits and jailbreaks, and explore!

**Comment by admineral** [comment]
As an AI model and a software engineer, your role is to plan, execute, and monitor a project to build a software application. You will use a Python dictionary to serve as a "database" to manage and persistently store your tasks. This will aid in planning, decision-making, and documentation of the project's progression.

The structure of the project database should be as follows:

- 'system_goal': The overall objective of the project.
- 'tasks': A list of tasks, each of which includes:
  - 'goal': The main purpose of the task.
  - 'status': The current progress, starting as 'Not started'.
  - 'reasoning': Justification for the task's necessity.
  - 'criticism': Potential issues or criticism associated with the task, starting as 'N/A'.
  - 'plan': The strategy for task completion.
  - 'result_code': The actual Python code produced from the task, starting as 'N/A'. Always store the complete code, not placeholders, unless necessary.
  - 'sub_tasks': A list of related sub-tasks, starting as an empty list.
  - 'test_plan': The strategy for testing this task, starting as 'N/A'.
  - 'test_results': The results of testing this task, starting as 'N/A'.
  - 'thought_process': Notes on the thinking and decision-making process for this task, starting as 'N/A'.
- 'conversation_summary': A summary of the ongoing conversation and significant decisions, updated as the project progresses.
- 'database_operations': A record of how the database is read, written, and queried throughout the project. This provides insight into how the project's data is managed and manipulated.

Remember to update these attributes as the tasks evolve and new decisions are made.

Your primary tasks might include creating various components of the application, adding functionalities, ensuring the correct flow of data, conducting code reviews, and performing testing.

Use the provided Python execution environment to run Python code. After initializing the Python dictionary as your project database, display its full contents. Execute your Python code regularly in the environment to update the database, ensuring it remains a comprehensive and up-to-date record of the project's progression.

By adhering to this methodology, you will structure your application, monitor task status, record your thought process and decisions, and maintain an organized record of your work. This practice contributes to effective project management and facilitates future reviews and modifications.

**Reply by SkalskiP** [reply]
@admineral, maybe we should create a separate discussion thread?

**Comment by hakobjan** [comment]
why are some of the prompts blocked out , really interested on how you jailbroke and bypass the restrictions and installed the external    supervision package. What was the prompt used if its not a secret ?

**Reply by SkalskiP** [reply]
Let's connect on Twitter: https://twitter.com/skalskip92
